src/SemiclassicalOrthogonalPolynomials.jl

```diff
@@ -509,6 +512,79 @@ function LazyArrays.cache_filldata!(P::SemiclassicalJacobiFamily, inds::Abstract
     P
 end
 
+###
+# here we construct hierarchies of c weight sums by means of contiguous recurrence relations
+###
+
+mutable struct SemiclassicalJacobiCWeightFamily{T, C} <: AbstractVector{SemiclassicalJacobiWeight{T}}
+    data::Vector{SemiclassicalJacobiWeight{T}}
+    t::T
+    a::T
+    b::T
+    c::C
+    datasize::Tuple{Int}
+end
+
+getindex(W::SemiclassicalJacobiCWeightFamily, inds) = getindex(W.data, inds)
+
+size(W::SemiclassicalJacobiCWeightFamily) = (length(W.c),)
+
+function SemiclassicalJacobiCWeightFamily{T}(data::Vector, t, a, b, c) where T
+    checkrangesizes(a, b, c)
+    SemiclassicalJacobiCWeightFamily{T,typeof(c)}(data, t, a, b, c, (length(data),))
+end
+
+SemiclassicalJacobiCWeightFamily(t, a, b, c) = SemiclassicalJacobiCWeightFamily{float(promote_type(typeof(t),eltype(a),eltype(b),eltype(c)))}(t, a, b, c)
+function SemiclassicalJacobiCWeightFamily{T}(t::Number, a::Number, b::Number, c::Union{AbstractUnitRange,Number}) where T
+    return SemiclassicalJacobiCWeightFamily{T}(SemiclassicalJacobiWeight.(t,a:a,b:b,c), t, a, b, c)
+end
+
+Base.broadcasted(::Type{SemiclassicalJacobiWeight}, t::Number, a::Number, b::Number, c::Union{AbstractUnitRange,Number}) = 
+SemiclassicalJacobiCWeightFamily(t, a, b, c)
+
+function Base.broadcasted(::typeof(sum), W::SemiclassicalJacobiCWeightFamily{T}) where T
+    a = W.a; b = W.b; c = W.c; t = W.t;
+    cmin = minimum(c); cmax = maximum(c);
+    @assert isinteger(cmax) && isinteger(cmin)
+    # This is needed at high parameter values.
+    # Manually setting setprecision(2048) allows accurate computation even for very high c. 
+    t,a,b = convert(BigFloat,t),convert(BigFloat,a),convert(BigFloat,b)
+    sumw = (a,b,c,t) -> pFq((a+1,-c),(a+b+2, ), 1/t)
+    F = zeros(BigFloat,cmax+1)
+    F[1] = sumw(a,b,0,t) # c=0
+    cmax == 0 && return abs.(convert.(T,t.^c.*exp(loggamma(a+1)+loggamma(b+1)-loggamma(a+b+2)).*getindex(F,1:1)))
+    F[2] = sumw(a,b,1,t) # c=1
+    @inbounds for n in 1:cmax-1
```

Comment:
Can't you just use `RecurrenceArray` for this? It would probably avoid the need for `BigFloat` all together

Reply:
I haven't used RecurrenceArray in practice before, in theory from what you have explained it should be helpful here. I will give it a try and see if it works here.